scheme/proto/src/main/java/cz/o2/proxima/scheme/proto/ProtoMessageValueAccessor.java

```diff
@@ -71,25 +71,35 @@ public ProtoMessageValueAccessor(Factory<T> defaultValueFactory) {
 
   @Override
   public StructureValue valueOf(T object) {
+    /*
+     Here we get all fields from descriptor which also returns empty fields (not set), those
+     fields needs to be filtered -> see filter on stream, where we pass all non-message and
+     non-repeated fields, repeated fields just when count > 0 and messages just when was set.
+    */
     return StructureValue.of(
         object
-            .getAllFields()
-            .entrySet()
+            .getDescriptorForType()
+            .getFields()
             .stream()
+            .filter(
+                f ->
+                    (!f.getJavaType().equals(JavaType.MESSAGE) && !f.isRepeated())
+                        || (!f.isRepeated() && object.hasField(f))
```

Comment:
Could we simplify this condition a bit? We could pull the `isRepeated` out like  `!f.isRepeated() && (MESSAGE || hasField(f)) || f.isRepeated() && getRepeatedCount`

Reply:
Changed.